src/Arcus.EventGrid.Proxy.Api/Controllers/v1/HealthController.cs

```diff
@@ -8,17 +12,32 @@ namespace Arcus.EventGrid.Proxy.Api.Controllers.v1
     [ApiController]
     public class HealthController : ControllerBase
     {
+        private readonly HealthCheckService _healthCheckService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HealthController"/> class.
+        /// </summary>
+        /// <param name="healthCheckService">The service to provide the health of the API application.</param>
+        public HealthController(HealthCheckService healthCheckService)
+        {
+            Guard.NotNull(healthCheckService, nameof(healthCheckService));
+
+            _healthCheckService = healthCheckService;
+        }
+
         /// <summary>
         ///     Get Health
         /// </summary>
         /// <remarks>Provides an indication about the health of the runtime</remarks>
         [HttpGet]
         [SwaggerOperation(OperationId = "Health_Get")]
-        [SwaggerResponse((int) HttpStatusCode.OK, Description = "Runtime is up and running in a healthy state")]
-        [SwaggerResponse((int) HttpStatusCode.ServiceUnavailable, Description = "Runtime is not healthy")]
-        public IActionResult Get()
+        [SwaggerResponse((int)HttpStatusCode.OK, Description = "Runtime is up and running in a healthy state", Type = typeof(HealthReport))]
+        [SwaggerResponse((int)HttpStatusCode.ServiceUnavailable, Description = "Runtime is not healthy")]
+        public async Task<IActionResult> Get()
         {
-            return Ok();
+            HealthReport healthReport = await _healthCheckService.CheckHealthAsync();
+
+            return healthReport?.Status == HealthStatus.Healthy ? Ok(healthReport) : StatusCode(StatusCodes.Status503ServiceUnavailable, healthReport);
```

Comment:
Maybe split in simpler `if`. We're in Arcus 😄 .

Reply:
👌